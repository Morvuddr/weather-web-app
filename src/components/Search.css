.search{
    display: flex;
    flex-direction: column;
    margin: 10px auto 0 auto;
    height: 160px;
    width: 90%;
    justify-content: space-between;
    align-items: center;
}
.search h1 {
    text-align: left;
}
.searchInput {
    height: 40px;
    width: 160px;
}

@media( min-width: 700px) {
    .search {
        height: 80px;
        flex-direction: row;
    }
}