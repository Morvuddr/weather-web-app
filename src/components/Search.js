import React from 'react';
import * as PropTypes from 'prop-types';
import '../styles/FavoritesContainer.css'

const Search = ({ onSubmit }) => (
    <div className='search'>
        <h1>
            Избранное
        </h1>
        <form onSubmit={onSubmit}>
            <input className='searchInput' placeholder='Добавить новый город' name='cityName' required='required'></input>
            <button className='searchButton' type='submit'>&#43;</button>
        </form>
    </div>
)

Search.propTypes = {
    onSubmit: PropTypes.func
}

export default Search;
