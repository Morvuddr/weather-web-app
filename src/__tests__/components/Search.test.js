import Search from '../../components/Search';
import renderer from 'react-test-renderer';
import React from 'react';

describe('Search component', () => {
    test('matches the snapshot', () => {
        const tree = renderer.create(
            <Search />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});