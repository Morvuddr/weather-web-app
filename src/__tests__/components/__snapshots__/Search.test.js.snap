// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component matches the snapshot 1`] = `
<div
  className="search"
>
  <h1>
    Избранное
  </h1>
  <form
    onSubmit={[Function]}
  >
    <input
      className="searchInput"
      name="cityName"
      placeholder="Добавить новый город"
      required="required"
    />
    <button
      className="searchButton"
      type="submit"
    >
      +
    </button>
  </form>
</div>
`;
